src/repl.ts

```diff
@@ -125,7 +132,15 @@ export function createRepl(options: CreateReplOptions = {}) {
     }
 
     try {
-      result = evalCode(code);
+      const { awaitPromise, result: evalResult } = evalCodeInternal(code);
+
+      if (awaitPromise) {
+        return (async () => {
+          callback(err, await evalResult);
+        })();
```

Comment:
`err` is null in this case (as it's defined), but I think I get your point, you want this instead?

```typescript
return (async () => {
  try {
    const awaitedEvalResult = await evalResult;
    callback(err, awaitedEvalResult);
  } catch (promiseError) {
    callback(promiseError, result)
  }
})();

Reply:
What line of code is declaring `err`?  I see `callback(err` but I don't see where `err` is declared.

Reply:
If `await evalResult` throws a `TSError` or a recoverable error, are those being handled correctly?  I see we have a `} catch (error) {` block, but an asynchronously-thrown `TSError` does not appear to go through that codepath.

If someone types `import ...` or `require()` into the repl, and it loads *another* TS file that throws a `TSError`, we are already catching those with the `} catch(error) {` block.  We should match when async, so that sync and async behave the same.

Reply:
These are declared right above:
 https://github.com/TypeStrong/ts-node/blob/40d78e174a5e36a43e3b40d2a85757934457ef95/src/repl.ts#L125-L141

Reply:
check last implementation, should resolve this issue.